docs/data/ampli/plugin.md

```diff
@@ -2,38 +2,209 @@
 
 Add plugin to Ampli via `ampli.client.add()`. You can add as many plugin as you like. Each plugin runs in the order based on the plugin type.
 
-=== "Typescript"
-    ```js
+=== "Typescript/Javascript"
+    ```ts
     ampli.client.add(yourPlugin())
     ```
 
+## Destination type plugin
+
+Here's an example of a plugin that sends each event that's instrumented to a target server URL using your preferred HTTP client.
+
 === "Javascript"
+
     ```js
-    ampli.client.add(yourPlugin())
+    export class MyDestinationPlugin {
+      name = 'my-destination-plugin
+      type = PluginType.DESTINATION
+
+      constructor(serverUrl) {
+        this.serverUrl = serverUrl
+      }
+
+      /**
+       * setup() is called on plugin installation
+       * example: client.add(new AddEventIdPlugin());
+       */
+      setup(config) {
+        this.amplitudeConfig = config
+      }
+
+      /**
+       * execute() is called on each event instrumented
+       * example: client.track('New Event');
+       */
+      execute(event) {
+        const payload = {
+          key: 'secret',
+          data: event
+        }
+        return fetch(this.serverUrl, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            Accept: "*/*"
+          },
+          body: JSON.stringify(payload)
+        }).then(function(response) {
+          return {
+            code: response.status,
+            event: event,
+            message: response.statusText
+          }
+        })
+      }
+    }
     ```
 
-## Plugin examples
+=== "Typescript"
+
+    ```ts
+    export class MyDestinationPlugin implements DestinationPlugin {
+      name = 'my-destination-plugin';
+      type = PluginType.DESTINATION as const;
+      amplitudeConfig: Config;
+      serverUrl: string;
+
+      constructor(serverUrl: string) {
+        this.serverUrl = serverUrl;
+      }
 
-### Enrichment type plugin
+     /**
+       * setup() is called on plugin installation
+       * example: client.add(new AddEventIdPlugin());
+       */
+      setup(config) {
+        this.amplitudeConfig = config;
+      }
+
+      /**
+       * execute() is called on each event instrumented
+       * example: client.track('New Event');
+       */
+      execute(event) {
+        const payload = {
+          key: 'secret',
+          data: event,
+        };
+        return fetch(this.serverUrl, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            Accept: '*/*',
+          },
+          body: JSON.stringify(payload),
+        }).then(function(response) {
+          return {
+            code: response.status,
+            event: event,
+            message: response.statusText,
+          };
+       });
+      }
+    }
+    ```
+
+### Hotjar Example
```

Comment:
Might be helpful to explain what the Hotjar plugin does, for example

The following code snippet provides a customized Amplitude destination plugin example that can be used to send events to Hotjar by using [hotjar browser SDK](https://www.npmjs.com/package/@hotjar/browser)